scipy/stats/_axis_nan_policy.py

```diff
@@ -268,12 +271,12 @@ def _check_empty_inputs(samples, axis):
     return output
 
 
-def _add_reduced_axes(res, reduced_axes, keepdims):
+def _add_reduced_axes(res, reduced_axes, keepdims, xp=np):
```

Comment:
The changes in this file are pretty straightforward xp-translations that allow non-numpy arrays to go through the first part of the decorator, which makes a few things much easier for wrapped functions.
  - The dimensions specified by `axis=None` and `axis` tuples are raveled (so functions only have to consider scalar, in `axis`)
  - The remaining `axis` is moved to `-1`, which tends to make indexing along it easier.
  - `keepdims` is supported automatically by adding back any axes that get reduced.

Eventually, I'll let non-numpy arrays pass a little further through the decorator so when the input array is too small (e.g. empty), the right thing happens.